### PR TITLE
gh-116789: Add more tests for `inspect.getmembers`

### PR DESCRIPTION
**Pull request by sobolevn**
It basically checks that `getmembers` is equal `dir(...) + getattr` for these types.
I've covered all types that were mentioned in the docs.

* Issue: gh-116789

### CODE REVIEW

File: Lib/test/test_inspect/test_inspect.py [hunk 1]
```diff
@@ -1512,6 +1512,71 @@ def f(self):
         self.assertIn(('f', b.f), inspect.getmembers(b))
         self.assertIn(('f', b.f), inspect.getmembers(b, inspect.ismethod))
 
+    def test_getmembers_function(self):
+        def func(arg: int = 1, *, kw = 2) -> None:
+            """Docstring."""
+
+        res = dict(inspect.getmembers(func))
```
Comment (serhiy-storchaka) [inline Lib/test/test_inspect/test_inspect.py]: Why call dict?

* It loses information about the result type of getmembers().
* It loses information about the order of the list.
* It loses information about possible duplicates.

**Review by serhiy-storchaka — COMMENTED** [review]
All these tests are just copies of each other. They do not contain anything specific to the specific type, so why add them? A test for arbitrary user object should be enough. They only depend on the result of dir(), if it returns an empty list, all tests will automatically pass.

File: Lib/test/test_inspect/test_inspect.py (hunk 1 above)
Comment (sobolevn) [reply Lib/test/test_inspect/test_inspect.py]: I agree, these tests can be much better, thanks a lot for the input 👍

Comment (sobolevn) [reply Lib/test/test_inspect/test_inspect.py]: I will rework them a bit later, not feeling well today.

Comment (vstinner) [reply Lib/test/test_inspect/test_inspect.py]: What's the status of this PR? Do you still plan to update it?

Comment (sobolevn) [reply Lib/test/test_inspect/test_inspect.py]: Yes, will do tomorrow!

**Comment by sobolevn** [comment]
Sorry, I lost this PR in the notifications at some point.

I addressed @serhiy-storchaka's feedback:
- I added two classes: one for correct `__dir__` and one for `__dir__` with extra item that does not exist
- I no longer use `dict`, checks are made specific
- Extra tests are removed

Thanks a lot for the feedback and notification! 👍

File: Lib/test/test_inspect/test_inspect.py [hunk 2]
```diff
@@ -1550,6 +1550,41 @@ def f(self):
...
+        cd = CorrectDir(5)
+        self.assertEqual(inspect.getmembers(cd), [
+            ('attr', 5),
+            ('method', cd.method),
+        ])
+        self.assertEqual(inspect.getmembers(cd, inspect.ismethod), [
+            ('method', cd.method),
+        ])
+
+    def test_getmembers_custom_broken_dir(self):
```
Comment (vstinner) [inline Lib/test/test_inspect/test_inspect.py:1571]: Can you add a comment explaining the purpose of this test?

**Review by vstinner — APPROVED** [review]
LGTM.

@serhiy-storchaka: Do you want to review the updated new tests?

**Comment by sobolevn** [comment]
It was up for 2 weeks, so I am going to merge this, we can add more tests later. Thanks a lot for the reviews, they were very helpful.

**Pull request merged**